### ISSUE TITLE
All my Leaves/Pending Leaves empty

### ISSUE THREAD

**Issue by gamonoid** [gamonoid/issues#37]
**[Original report](https://bitbucket.org/thilina/icehrm-opensource/issue/37) by Anonymous.**

----------------------------------------

Hi, Thilina.
One of my users has sent 3 pending leaves, but his "All my Leaves" and "Pending Leaves" tabs are empty. He can't even see the search box or the leaves list, it's empty. But if he goes to "Leave Calendar" he can see his leaves in the calendar.
No errors found in log files. Only appeared the following in Firefox (F12): getPreventDefault() is disapproved. Use defaultPrevented instead. jquery.js:3257
TypeError: data is null AdapterBase.js:148
The environment is LAMP on Ubuntu.
Thanks in advance

**Comment by gamonoid** [comment]
**Original comment by Thilina Hasantha (Bitbucket: [thilina](https://bitbucket.org/thilina), GitHub: [thilina](https://github.com/thilina)).**

----------------------------------------

Hi, did you try switching to that user and checking the two tabs which are not working.

Also check following directives in php.ini. All three should be off


```
#!php

magic_quotes_gpc = Off
magic_quotes_runtime = Off
magic_quotes_sybase = Off
```

**Comment by gamonoid** [comment]
**Original comment by Francisco Quevedo (Bitbucket: [itvecisa](https://bitbucket.org/itvecisa), GitHub: Unknown).**

----------------------------------------

I tried with my own laptop in different browsers login as those 2 users and login as admin and switching to them with the same results.
I have written the directives in php.ini and restarted apache but still the two tabs are empty.

**Comment by gamonoid** [comment]
**Original comment by Francisco Quevedo (Bitbucket: [itvecisa](https://bitbucket.org/itvecisa), GitHub: Unknown).**

----------------------------------------

Hi, Thilina.
Maybe is not related, but I see that in IE the response is empty and there is the following error:
SCRIPT5007: Unable to get property 'status'of undefined or null reference
AdapterBase.js?v=41.0, Line148 Character 3
Thanks a lot

**Comment by gamonoid** [comment]
**Original comment by Thilina Hasantha (Bitbucket: [thilina](https://bitbucket.org/thilina), GitHub: [thilina](https://github.com/thilina)).**

----------------------------------------

Can you check the icehrm error log. Path to error log can be found in first line of app/config.php

If required change this path to an allowed path in your server.

After making the request which is failing place let me know if there is any error getting printed

**Comment by gamonoid** [comment]
**Original comment by Thilina Hasantha (Bitbucket: [thilina](https://bitbucket.org/thilina), GitHub: [thilina](https://github.com/thilina)).**

----------------------------------------

Also make sure to comment out line 2 in include.common.php


```
#!php

<?php
error_reporting(E_ERROR); 
```

Should be 


```
#!php

<?php
//error_reporting(E_ERROR);
```

**Comment by gamonoid** [comment]
**Original comment by Francisco Quevedo (Bitbucket: [itvecisa](https://bitbucket.org/itvecisa), GitHub: Unknown).**

----------------------------------------

It was commented out. Debuging in Chrome I see the request and the response in both cases is empty.

**Comment by gamonoid** [comment]
**Original comment by Francisco Quevedo (Bitbucket: [itvecisa](https://bitbucket.org/itvecisa), GitHub: Unknown).**

----------------------------------------

I paste the log:
[13-May-2014 08:07:22 UTC] PHP Notice:  Use of undefined constant MODULE_PATH - assumed 'MODULE_PATH' in /var/www/icehrm/server.includes.inc.php on line 10
[13-May-2014 08:07:22 UTC] PHP Notice:  Undefined offset: -1 in /var/www/icehrm/server.includes.inc.php on line 12
[13-May-2014 08:07:22 UTC] PHP Notice:  Undefined index: Time Management in /var/www/icehrm/modules.php on line 163
[13-May-2014 08:07:22 UTC] PHP Notice:  Undefined index: Documents in /var/www/icehrm/modules.php on line 163
[13-May-2014 08:07:22 UTC] PHP Notice:  Undefined index: Loans in /var/www/icehrm/modules.php on line 163
[13-May-2014 08:07:32 UTC] PHP Warning:  strtotime(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in /var/www/icehrm/modules/employees/api/EmployeesActionManager.php on line 37
[13-May-2014 08:07:32 UTC] PHP Warning:  date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in /var/www/icehrm/modules/employees/api/EmployeesActionManager.php on line 37
[13-May-2014 08:07:32 UTC] PHP Warning:  strtotime(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in /var/www/icehrm/modules/employees/api/EmployeesActionManager.php on line 45
[13-May-2014 08:07:32 UTC] PHP Warning:  date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in /var/www/icehrm/modules/employees/api/EmployeesActionManager.php on line 45
[13-May-2014 08:07:57 UTC] PHP Notice:  Undefined index: Time Management in /var/www/icehrm/modules.php on line 163
[13-May-2014 08:07:57 UTC] PHP Notice:  Undefined index: Documents in /var/www/icehrm/modules.php on line 163
[13-May-2014 08:07:57 UTC] PHP Notice:  Undefined index: Loans in /var/www/icehrm/modules.php on line 163

**Comment by gamonoid** [comment]
**Original comment by Francisco Quevedo (Bitbucket: [itvecisa](https://bitbucket.org/itvecisa), GitHub: Unknown).**

----------------------------------------

Setting the date.timezone var in php.ini the date() and strtotime() dissappeared. The error remains.

**Comment by gamonoid** [comment]
**Original comment by Thilina Hasantha (Bitbucket: [thilina](https://bitbucket.org/thilina), GitHub: [thilina](https://github.com/thilina)).**

----------------------------------------

This is fixed in v5.0 which can be downloaded from https://sourceforge.net/projects/icehrm/files/latest/download
You need to set [System: Do not pass JSON in request] = Yes in Settings module
